packages/router/src/router.tsx

```diff
@@ -251,9 +258,12 @@ const NotFoundChecker: React.FC<{ children: React.ReactNode }> = ({
         )
 
         if (match) {
+          // name will be undefined for redirect routes
+          routeName = name || ''
```

Comment:
🤷‍♂️ 
```suggestion
          routeName = name || '' // `name` is undefined for redirect routes
```

Reply:
I've changed this code, hopefully it's not as confusing anymore

Reply:
Nice!